### PR TITLE
fix intermittent failure to start unit test runner

### PR DESCRIPTION
**Pull request by MoserMichael**
Occasionally ginkgo test run fails to start, the error message in this
case is 'unable to combine coverage'
This condition is fixed by repeating the test run.

The full error log for a failed test run:

      quorum broken
      /home/mmoser/go/src/kubevirt.io/node-maintenance-operator/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:43
    [It] quorum broken
      /home/mmoser/go/src/kubevirt.io/node-maintenance-operator/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:43
    path is /home/mmoser/go/src/kubevirt.io/node-maintenance-operator
    Unable to read coverage file pkg/controller/nodemaintenance/cover.out to combine, open pkg/controller/nodemaintenance/cover.out: no such file or directory

Signed-off-by: Michael Moser <mmoser@redhat.com>

### CODE REVIEW

File: hack/coverage.sh [hunk 1]
```diff
@@ -13,9 +14,31 @@ declare -a EXCLUDE_FILES_FROM_COVERAGE=("nodemaintenance_controller_init.go")
...
 
-# run ginkgo with coverage result line
-${GINKGO} ${GINKGO_ARGS} ./pkg/ ./cmd/ | sed  '/coverage:.*$/d'
-GSTAT=${PIPESTATUS[0]}
+REPEAT_COUNT=0
+MAX_REPEAT_COUNT=10
+while [[ $REPEAT_COUNT -lt $MAX_REPEAT_COUNT ]]; do
+
+		# run ginkgo with coverage result line
+		${GINKGO} ${GINKGO_ARGS} ./pkg/ ./cmd/ 2>&1 | sed  '/coverage:.*$/d'  | tee msg
```
Comment (n1r1) [inline hack/coverage.sh:22]: Instead of using `tee` you might want just to use `grep`, this will be helpful in order to avoid creating new files on disks that later could appear as changes in git

Comment (MoserMichael) [reply hack/coverage.sh]: fixed.

File: hack/coverage.sh [hunk 2]
```diff
@@ -4,7 +4,8 @@ GINKGO="$1"
 TARGETCOVERAGE="$2"
 
 COVERAGE_FILE=cover.out
-GINKGO_COVERAGE_ARGS="-cover -coverprofile=${COVERAGE_FILE} -outputdir=. --skipPackage ./vendor"
+
+GINKGO_COVERAGE_ARGS="-cover -coverprofile=${COVERAGE_FILE} -outputdir=.  --skipPackage ./vendor"
```
Comment (beekhof) [inline hack/coverage.sh:8]: Can we drop the unnecessary whitespace changes please?

File: hack/coverage.sh [hunk 3]
```diff
@@ -13,9 +14,32 @@ declare -a EXCLUDE_FILES_FROM_COVERAGE=("nodemaintenance_controller_init.go")
...
 
-# run ginkgo with coverage result line
-${GINKGO} ${GINKGO_ARGS} ./pkg/ ./cmd/ | sed  '/coverage:.*$/d'
-GSTAT=${PIPESTATUS[0]}
+REPEAT_COUNT=0
+MAX_REPEAT_COUNT=10
+while [[ $REPEAT_COUNT -lt $MAX_REPEAT_COUNT ]]; do
+
+		# run ginkgo with coverage result line
+		GINKGO_OUT=$(${GINKGO} ${GINKGO_ARGS} ./pkg/ ./cmd/ 2>&1 | sed  '/coverage:.*$/d')
```
Comment (beekhof) [inline hack/coverage.sh:22]: Are you sure that the `coverage:` lines do not contain any information about why this test is flaky?

Comment (MoserMichael) [reply hack/coverage.sh:22]: The lines taken out are from nodemaintenace_controller_init.go - these are functions that have zero coverage during unit testing. (like addition of controller to the controller runtime manager etc).
I checked that the problem happened without these changes at all, so i don't think there is a connection.

File: hack/coverage.sh (hunk 2 above)
Comment (MoserMichael) [reply hack/coverage.sh]: fixed

File: hack/coverage.sh (hunk 3 above)
Comment (MoserMichael) [reply hack/coverage.sh:21]: i get it without this trick too; see https://github.com/onsi/ginkgo/issues/417

Comment (MoserMichael) [reply hack/coverage.sh:21]: I actually haven't seen these failures of the test runner for quite some time now.
It might have been some  problem on the side of the build server environment.

**Comment by beekhof** [comment]
Problem seems to have gone away
